### PR TITLE
Migrates lots of StorageProviderBase refs to UnifiedStore

### PR DESCRIPTION
**Pull request by csilvestrini**
There's only a couple of trickier usages of StorageProviderBase left to migrate (plus all the references in the old storage code, which will be deleted anyway).

### CODE REVIEW

File: src/runtime/api-channel.ts [hunk 1]
```diff
@@ -483,16 +482,16 @@ export abstract class PECOuterPort extends APIPort {
...
+  abstract onHandleGet(handle: UnifiedStore, callback: number);
+  abstract onHandleToList(handle: UnifiedStore, callback: number);
+  abstract onHandleSet(handle: UnifiedStore, data: {}, particleId: string, barrier: string);
+  abstract onHandleClear(handle: UnifiedStore, particleId: string, barrier: string);
+  abstract onHandleStore(handle: UnifiedStore, callback: number, data: {value: {}, keys: string[]}, particleId: string);
+  abstract onHandleRemove(handle: UnifiedStore, callback: number, data: {}, particleId: string);
+  abstract onHandleRemoveMultiple(handle: UnifiedStore, callback: number, data: {}, particleId: string);
+  abstract onHandleStream(handle: UnifiedStore, callback: number, pageSize: number, forward: boolean);
+  abstract onStreamCursorNext(handle: UnifiedStore, callback: number, cursorId: number);
+  abstract onStreamCursorClose(handle: UnifiedStore, cursorId: number);
```
Comment (galganif) [inline src/runtime/api-channel.ts]: I think all of these are obsoleted by onProxyMessage.

Comment (csilvestrini) [reply src/runtime/api-channel.ts]: Thanks, done! Added comment too

File: src/runtime/tests/handle-test.ts [hunk 2]
```diff
@@ -12,10 +12,9 @@ import {Arc} from '../arc.js';
...
 import {Manifest} from '../manifest.js';
-import {Schema} from '../schema.js';
-import {CollectionStorageProvider, SingletonStorageProvider, StorageProviderBase} from '../storage/storage-provider-base.js';
+import {CollectionStorageProvider, SingletonStorageProvider} from '../storage/storage-provider-base.js';
 import {EntityType, InterfaceType} from '../type.js';
-import {Id, ArcId, IdGenerator} from '../id.js';
+import {ArcId, IdGenerator} from '../id.js';
 import {NoOpStorageProxy} from '../storage-proxy.js';
 
 describe('Handle', () => {
```
[the remark this answers is not in the document]
Comment (csilvestrini) [reply src/runtime/tests/handle-test.ts]: Reverted

File: src/runtime/tests/arc-test.ts [hunk 3]
```diff
@@ -53,7 +54,7 @@ async function setup(storageKeyPrefix: string) {
   };
 }
 
-function getSingletonHandle(store: StorageProviderBase): Singleton {
+function getSingletonHandle(store: UnifiedStore): Singleton {
```
[the remark this answers is not in the document]
Comment (csilvestrini) [reply src/runtime/tests/arc-test.ts]: I'll leave this as-is for now (it'll make it easier for me to see what code has been "updated" and what hasn't. I'll have to come back to it when I get to Singleton

**Pull request merged**